File: app/src/main/java/com/komsi/maleshop/adapter/CartAdapter.java
package com.komsi.maleshop.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.komsi.maleshop.model.Produk;
import com.komsi.maleshop.R;

import java.util.ArrayList;

public class CartAdapter extends RecyclerView.Adapter<CartAdapter.ViewHolder> {

    private Context context;
    private ArrayList<Produk> list;
    private TextView hargaProduk;
    private Double hargaTotal = 0.0;


    public CartAdapter(Context context, ArrayList<Produk> list, TextView hargaProduk) {
        this.context = context;
        this.list = list;
        this.hargaProduk = hargaProduk;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {

        View view = LayoutInflater.from(context).inflate(R.layout.cart_item, parent, false);

        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull final ViewHolder holder, final int position) {

        Glide.with(context).load(list.get(position).getThumbnail()).into(holder.imgCart);
        holder.cartTitle.setText(list.get(position).getNama());
        holder.cartColor.setText(list.get(position).getColour());
        holder.cartPrize.setText(list.get(position).getHarga());
        hargaProduk.setText("Rp " + getHargaTotal());
        holder.imgRemove.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View view) {
                list.remove(position);
                notifyItemRemoved(position);
                hargaProduk.setText("Rp " + getHargaTotal());
                notifyItemRangeChanged(position,list.size());
            }
        });
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        private TextView cartTitle;
        private TextView cartColor;
        private TextView cartSize;
        private TextView cartPrize;
        private ImageView imgCart;
        private Button imgRemove;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);

            imgCart = itemView.findViewById(R.id.img_cart);
            cartTitle = itemView.findViewById(R.id.tv_nama_cart);
            cartSize = itemView.findViewById(R.id.tv_ukuran_cart);
            cartPrize = itemView.findViewById(R.id.tv_harga_cart);
            cartColor = itemView.findViewById(R.id.tv_warna_cart);
            imgRemove = itemView.findViewById(R.id.icon_remove_cart);

        }
    }

    public Double getHargaTotal(){
        hargaTotal = 0.0;
        for (int i=0;i<list.size();i++){
            hargaTotal += Double.parseDouble(list.get(i).getHarga());
        }
        return hargaTotal;

    }
}

File: app/src/main/java/com/komsi/maleshop/adapter/CartAdapter.kt
package com.komsi.maleshop.adapter

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import com.androidnetworking.AndroidNetworking
import com.androidnetworking.common.Priority
import com.androidnetworking.error.ANError
import com.androidnetworking.interfaces.StringRequestListener
import com.bumptech.glide.Glide
import com.komsi.maleshop.R
import com.komsi.maleshop.model.Cart
import com.komsi.maleshop.persistence.Credential
import com.komsi.maleshop.utils.ConstApi
import java.util.*

class CartAdapter(private val context: Context,private val callback:ProductCallback) : RecyclerView.Adapter<CartAdapter.ViewHolder>() {
    val list = ArrayList<Cart>()

    fun setData(item: ArrayList<Cart>) {
        this.list.clear()
        this.list.addAll(item)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.item_cart, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        Glide.with(context).load(ConstApi.PRODUCT_IMAGE_URL.value + list[position].foto).into(holder.imgCart)
        holder.cartTitle.text = list[position].nama
        holder.cartPrize.text = list[position].harga.toString()
        holder.tvQty.text = list[position].qty

        holder.imgRemove.setOnClickListener {
            deteteCartItem(list[position].id.toString(),position)
        }
    }

    private fun deteteCartItem(id: String, position: Int) {
        AndroidNetworking.initialize(context)
        AndroidNetworking.delete(ConstApi.CART.value+ "/$id")
                .addHeaders("Authorization", "Bearer ${Credential.getToken(context)}")
                .addBodyParameter("produk_id", id)
                .setTag("favorite")
                .setPriority(Priority.HIGH)
                .build()
                .getAsString(object : StringRequestListener {
                    override fun onResponse(response: String?) {
                        Toast.makeText(context, "Success", Toast.LENGTH_SHORT).show()
                        list.removeAt(position)
                        notifyItemRemoved(position)
                        notifyItemChanged(position)
                    }

                    override fun onError(anError: ANError?) {
                        Toast.makeText(context, "Error ${anError.toString()}", Toast.LENGTH_SHORT).show()
                        Log.d("favResponse",anError.toString())
                    }
                })
    }

    override fun getItemCount(): Int {
        return list.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val cartTitle: TextView = itemView.findViewById(R.id.tv_nama_cart)
        val cartPrize: TextView = itemView.findViewById(R.id.tv_harga_cart)
        val imgCart: ImageView = itemView.findViewById(R.id.img_cart)
        val imgRemove: ImageView = itemView.findViewById(R.id.icon_remove_cart)
        val tvQty: TextView = itemView.findViewById(R.id.tv_qty)
    }

}